### DISCUSSION TITLE
How to do data analysis in EMQX for specific topic

### DISCUSSION THREAD

**Discussion by dineshwar21** [emqx/emqx#13781] [Q&A] [answered]
I'm currently analyzing the number of requests a specific user makes to a particular topic. To achieve this, I plan to use log traces with a 3-month log rotation. However, I noticed that downloading a log file of around 140 MB via the **API (GET /api/v5/trace/{name}/download)** takes more than 40 seconds.

I'm looking for a better approach to perform this data analysis without relying on external services. Is there a way to optimize or minimize the log trace size to improve performance? Any advice or recommendations on improving the efficiency of this process would be greatly appreciated.

**Comment by zmstone** [comment, marked as answer]
You can always (ssh) login to the EMQX runner host and download the file directly.
The files are `data/trace/*.log` in each EMQX node

**Reply by dineshwar21** [reply]
My log trace located here _/var/lib/emqx/trace_ it's the right path?

Will the log file has created date with it?

![image](https://github.com/user-attachments/assets/beff6256-f600-4b93-b8b4-689edee05ee6)

I'm planning to do python script to fetch the data, just want to confirm. 

- Is there a way to retrieve the exact file name?
- Will the [PUBLISH] message always contain the username if provided? (I am using EMQX V5.7.2)

**Reply by zmstone** [reply]
The file name format is `trace_{NAME}_{DATE}`, where NAME is your input, and DATE is the date when the traces started (the `start_at` field)
However, this format may change in the future as it's very much internal.
A safer bet is to list the traces with API call: https://docs.emqx.com/en/emqx/v5.8/admin/api-docs.html#tag/Trace/paths/~1trace/get

Yes [PUBLISH] will always contain username if provided.

**Reply by dineshwar21** [reply]
One thing to confirm 
The trace_{NAME} will continue to exist in future file names, correct? I have implemented logic to search for files in that directory based on a specific name.

@zmstone Thank you so much for all your help! I truly appreciate the time.

**Reply by zmstone** [reply]
Yes, this `trace_` prefix is unlikely to change.

**Comment by dineshwar21** [comment]
Great
Thank you once again for all your help.
